### ISSUE TITLE
cpufreq plugin: no values with intel_pstate driver

### ISSUE THREAD

**Issue by hhoffstaette** [collectd/collectd#624]
On Sandy Bridge and newer CPUs newer kernels can (and should) use the intel_pstate frequency driver instead of the ACPI driver . This driver has new/changed sysfs entries, so the currently read values at

```
/sys/devices/system/cpu/cpuX/cpufreq/scaling_cur_freq
```

no longer exist. The new values seems to be:

```
/sys/devices/system/cpu/cpuX/cpufreq/cpuinfo_cur_freq
```

Which values to read could be detected by the existence of the entry called

```
/sys/devices/system/cpu/intel_pstate/
```

**Comment by rubenk** [comment]
Is this still the case? Afaict this is fixed in later kernels by https://github.com/torvalds/linux/commit/c034b02e213d271b98c45c4a7b54af8f69aaac1e

Documentation/cpu-freq/user-guide.txt also hints that scaling_cur_freq and cpuinfo_cur_freq are not the same. The latter takes turbo boost into account.

**Comment by hhoffstaette** [comment]
I no longer use collectd but just verified that with 4.1 and intel_pstate `/sys/devices/system/cpu/cpuX/cpufreq/scaling_cur_freq` exists again, so apparently this has changed. If someone can confirm that it works correctly with the intel_pstate driver (which should be the automatically selected default for newer Intel CPUs) I'd be more than happy to close this.

**Comment by igsol** [comment]
Please, pay attention to https://github.com/collectd/collectd/pull/1730 .

I had to create those changes, cause original cpufreq plugin doesn't collect frequency on such Linux version:
```
$ uname -a
Linux cees-dtn0 3.10.0-327.el7.x86_64 #1 SMP Thu Nov 19 22:10:57 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -a
LSB Version:	:core-4.1-amd64:core-4.1-noarch
Distributor ID:	CentOS
Description:	CentOS Linux release 7.2.1511 (Core) 
Release:	7.2.1511
Codename:	Core
```

**Comment by rubenk** [comment]
This seems fixed in CentOS 7.3:
```
ruben@test1: ~$ uname -r
3.10.0-514.el7.x86_64
ruben@test1: ~$ cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq
2401000
```

**Comment by Poil** [comment]
Hi,

Sorry to reopen this, but, on CentOS 7.3.1611 I don't have /sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq but I have /sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_cur_freq

```
# cpupower  frequency-info
analyzing CPU 0:
  driver: intel_pstate
  CPUs which run at the same hardware frequency: 0
  CPUs which need to have their frequency coordinated by software: 0
  maximum transition latency:  Cannot determine or is not supported.
  hardware limits: 1.20 GHz - 2.90 GHz
  available cpufreq governors: performance powersave
  current policy: frequency should be within 2.90 GHz and 2.90 GHz.
                  The governor "performance" may decide which speed to use
                  within this range.
  current CPU frequency: 2.29 GHz (asserted by call to hardware)
  boost state support:
    Supported: yes
    Active: yes
```

CPU : Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz
